resolver/serviceResolver.go

```diff
@@ -231,6 +231,29 @@ func (resolver *serviceResolver) VerifyCode(userAddress sdkCore.AddressHandler,
 	return verifyCodeData, nil
 }
 
+// SignMessage validates user's message, then adds guardian signature and returns the message.
+func (resolver *serviceResolver) SignMessage(userAddress sdkCore.AddressHandler, request requests.SignMessage) ([]byte, error) {
+	addressBytes := userAddress.AddressBytes()
+	resolver.userCritSection.RLock(string(addressBytes))
+	userInfo, err := resolver.getUserInfo(addressBytes)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve user info: %w", err)
+	}
+
+	guardianCryptoHolder, err := resolver.cryptoComponentsHolderFactory.Create(userInfo.FirstGuardian.PrivateKey)
```

Comment:
This is not ok, because the received codes are never used + the guardian to test them against should be received on request

We should follow the structure of `SignTransaction` here because we also want to count the code verification failures and apply the same backoff flow, so we'll need a something similar to `validateTxRequestReturningGuardian`, more specifically we need the `checkAllowanceAndVerifyCode + `getGuardianForTx` calls

proposed changes:
- extend `SignMessage struct` to also receive the guardian address to check the codes against it
- on `guardianGroup.go`, also extract `userIp` + `userAgent`, as in `signTransaction`, further needed for backoff
- update `getGuardianForTx` to `getGuardianInfoFromAddress` in order to make it reusable for both `signMessage` and `signTransaction` (so first parameter can become a string, the guardian address). This method guarantees that the provided guardian is the one known by tcs, even though the initial checks are done on code verification.
- use the above returned guardian as parameter for `resolver.cryptoComponentsHolderFactory.Create`
- on `/sign-message` request, also return the `otpCodeVerifyData` resulted after codes verification, so it can be forwarded to the end user